### PR TITLE
To add brief intro for CPU backend optimization

### PR DESCRIPTION
**Pull request by ZailiWang**
This PR is about adding brief introduction for x86 CPU backend optimization. Per previous discussion, the former PR #103307 was closed and creating this one, the contents are put into a new file.
@Guobing-Chen @jgong5 @mingfeima @jingxu10 please help review, thanks.

### CODE REVIEW

**Review by jgong5 — APPROVED** [review]

File: docs/source/compile/best-practice-for-backends.rst [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+==========================
+
+x86 CPU
+-----------------
+
+Compiled workloads on modern x86 CPUs are usually optimized by SIMD instruction sets. SIMD (Single Instruction Multiple Data) is a typical parallel processing technique for high performance computing, such as deep learning model training and inference. With SIMD applied, each compute unit performs the same instruction with different allocated data at any given timeslot. The most commonly deployed x86 instruction set architectures (ISAs) enabling SIMD include `AVX, AVX2, AVX-512 <https://en.wikipedia.org/wiki/Advanced_Vector_Extensions>` and `AMX <https://en.wikipedia.org/wiki/Advanced_Matrix_Extensions>`.
+
+Which ISAs are supported by a machine can be checked with `the tutorial <https://linuxconfig.org/identify-if-my-cpu-using-32-bit-or-64-bit-instruction-set>`.
+
+Specifically, with a server having AMX instructions enabled, workloads performance can be further boosted by `leveraging AMX <https://pytorch.org/tutorials/recipes/recipes/amx.html>`.
```
[the remark this answers is not in the document]
Comment (ZailiWang) [reply docs/source/compile/best-practice-for-backends.rst:11]: Hi Vadim, the PR for AMX tutorial page [#2439](https://github.com/pytorch/tutorials/pull/2439) has just been merged and will be available soon.
The AMX tutorial page would be indexed here https://pytorch.org/tutorials/recipes/recipes/index.html. There are 2 layers of "recipes" in these links.

[the remark this answers is not in the document]
Comment (vadimkantorov) [reply docs/source/compile/best-practice-for-backends.rst:11]: "Single-layer recipes" is actually available right now: https://pytorch.org/tutorials/recipes/amx.html

[the remark this answers is not in the document]
Comment (ZailiWang) [reply docs/source/compile/best-practice-for-backends.rst]: Link corrected, thanks for pointing this out.

File: docs/source/compile/best-practice-for-backends.rst [hunk 2]
```diff
@@ -0,0 +1,11 @@
+Best Practice for Backends
+==========================
+
+x86 CPU
+-----------------
+
+Compiled workloads on modern x86 CPUs are usually optimized by SIMD instruction sets. SIMD (Single Instruction Multiple Data) is a typical parallel processing technique for high performance computing, such as deep learning model training and inference. With SIMD applied, each compute unit performs the same instruction with different allocated data at any given timeslot. The most commonly deployed x86 instruction set architectures (ISAs) enabling SIMD include `AVX, AVX2, AVX-512 <https://en.wikipedia.org/wiki/Advanced_Vector_Extensions>` and `AMX <https://en.wikipedia.org/wiki/Advanced_Matrix_Extensions>`.
+
+Which ISAs are supported by a machine can be checked with `the tutorial <https://linuxconfig.org/identify-if-my-cpu-using-32-bit-or-64-bit-instruction-set>`.
```
Comment (malfet) [inline docs/source/compile/best-practice-for-backends.rst:9]: Should we recommend `collect_env` instead?

**Review by malfet — APPROVED** [review]

File: docs/source/compile/best-practice-for-backends.rst (hunk 2 above)
Comment (ZailiWang) [reply docs/source/compile/best-practice-for-backends.rst]: Updated, thanks for the advice.

**Comment by ZailiWang** [comment]
Hi @svekars would you please help merge this PR? Thanks.

**Comment by kit1980** [comment]
> Hi @svekars would you please help merge this PR? Thanks.

@ZailiWang What help do you need?
The PR is approved, you can just post "@pytorchbot merge comment" to do the merge.

**Pull request closed without merging**